src/Ng/Jobs/NgJob.cs

```diff
@@ -58,6 +61,11 @@ public virtual string GetUsage()
             return GetUsageBase();
         }
 
+        public void SetSecretInjector(ISecretInjector secretInjector)
```

Comment:
Is it possible to pass this into the constructor?

Reply:
I think about this at the first time. But then I realize that only 3 jobs need this and if we change the constructor, we have to change every job's constructor.
Not feel strong about this, If you feel that passing to the constructor is better, I will just do that.

Reply:
Not a big deal. This is fine if you think it's more appropriate for the situation.